src/all/unionmangas/src/eu/kanade/tachiyomi/extension/all/unionmangas/UnionMangasDto.kt

```diff
@@ -1,149 +1,69 @@
 package eu.kanade.tachiyomi.extension.all.unionmangas
 
-import eu.kanade.tachiyomi.source.model.SChapter
 import eu.kanade.tachiyomi.source.model.SManga
 import kotlinx.serialization.SerialName
 import kotlinx.serialization.Serializable
 
 @Serializable
-class NextData<T>(val props: Props<T>, val query: QueryDto) {
-    val data get() = props.pageProps
-}
-
-@Serializable
-class Props<T>(val pageProps: T)
-
-@Serializable
-class PopularMangaProps(@SerialName("data_popular") val mangas: List<PopularMangaDto>)
-
-@Serializable
-class LatestUpdateProps(@SerialName("data_lastuppdate") val latestUpdateDto: MangaListDto)
+class MangaDetailsDto(private val data: Props) {
+    val details: MangaDto get() = data.details
 
-@Serializable
-class MangaDetailsProps(@SerialName("dataManga") val mangaDetailsDto: MangaDetailsDto)
-
-@Serializable
-class ChaptersProps(@SerialName("data") val pageListData: String)
-
-@Serializable
-abstract class Pageable {
-    abstract var currentPage: String?
-    abstract var totalPage: Int
-
-    fun hasNextPage() =
-        try { (currentPage!!.toInt() + 1) < totalPage } catch (_: Exception) { false }
+    @Serializable
+    class Props(
+        @SerialName("infoDoc") val details: MangaDto,
+    )
 }
 
 @Serializable
-class ChapterPageDto(
-    val totalRecode: Int = 0,
-    override var currentPage: String?,
-    override var totalPage: Int,
-    @SerialName("data") val chapters: List<ChapterDto> = emptyList(),
-) : Pageable() {
-    fun toSChapter(langOption: LanguageOption): List<SChapter> =
-        chapters.map { chapter ->
-            SChapter.create().apply {
-                name = chapter.name
-                date_upload = chapter.date.toDate()
-                url = "/${langOption.infix}${chapter.toChapterUrl(langOption.chpPrefix)}"
-            }
-        }
-
-    private fun String.toDate(): Long =
-        try { UnionMangas.dateFormat.parse(trim())!!.time } catch (_: Exception) { 0L }
-
-    private fun ChapterDto.toChapterUrl(prefix: String) = "/${this.slugManga}/$prefix-${this.id}"
+open class Pageable<T>(
+    var currentPage: Int,
+    var totalPage: Int,
+    val data: List<T>,
+) {
+    fun hasNextPage() =
+        try { (currentPage + 1) <= totalPage } catch (_: Exception) { false }
```

Comment:
what exception are you expceting here?

Reply:
I forgot to remove, no need to try/catch